src/doc/trpl/unsafe.md

```diff
@@ -127,6 +110,111 @@ The latter assumption allows the compiler to optimize more
 effectively. As can be seen, actually *creating* a raw pointer is not
 unsafe, and neither is converting to an integer.
 
+## Reference and pointer aliasing
+
+Several examples can give a better comprehension of the reference and pointer
+aliasing rules.
+
+First of all, using `unsafe` code to circumvent the restrictions on references
+leads to undefined behaviour. For example, the following code creates two
+aliasing `&mut` references, and is therefore illegal.
```

Comment:
This code was removed in this commit https://github.com/rust-lang/rust/commit/5048953debe9eb44c672f3c453e894bd64a1301a
But I suppose this code was correct, because all references pointed to `UnsafeCell` and `UnsafeCell` is special.

The statement "if you create two aliasing `&mut` references, then it's UB" was taken from the old text, I was conservative and kept it. But it may be too restrictiveve, for example, the alternative (better?) wording could be "it's not UB when you simply create two aliasing `&mut` references, UB happens when you start writing/reading through them to/from something that is not `UnsafeCell`".